common/djangoapps/student/models.py

```diff
@@ -1486,11 +1487,14 @@ def emit_event(self, event_name):
             }
             if event_name == EVENT_NAME_ENROLLMENT_ACTIVATED:
                 segment_properties['email'] = self.user.email
+                # This next property is for an experiment, see method's comments for more information
+                segment_properties['external_course_updates'] = set_up_external_course_updates(self.user,
+                                                                                               self.course_id)
             with tracker.get_tracker().context(event_name, context):
                 tracker.emit(event_name, data)
                 segment.track(self.user_id, event_name, segment_properties)
 
-        except:  # pylint: disable=bare-except
+        except Exception:  # pylint: disable=broad-except
```

Comment:
curious why make this change?

Reply:
Bare excepts are (generally) evil. They catch things like SystemExit (a call to `sys.exit()`) and KeyboardInterrupt (user pressing ctrl+c). You almost never (99.9999%) want that.

Broad excepts only catch "normal" exceptions (just a wide swath of them, which is often discouraged, but not bad in and of itself).

This was just a drive-by fix.

Reply:
good to know, thanks!